### PR TITLE
Provisioning fixes

### PR DESCRIPTION
**Pull request by chrisnojima**
- [x] better logging of responses
- [x] maybeCancelProvision used to submit a cancel and wait, but now it'll nav you all the way back
- [x] better logging when initial provision call finishes, marking singleton as done
- [x] final error page is always layered on top of root and not appended to so clicking back puts you back to the right place
- [x] use better cancel method. fixes clicking back on paperkey screen not exiting login flow

### CODE REVIEW

**Comment by chrisnojima** [comment]
@keybase/react-hackers @cjb @patrickxb

File: shared/constants/provision.js [hunk 1]
```diff
@@ -8,7 +8,7 @@ import HiddenString from '../util/hidden-string'
 export const waitingKey = 'provision:waiting'
 
 // Do NOT change this. This is the value used by the daemon also so this way we can ignore it when they do it / when we do
-export const cancelDesc = 'kex canceled by caller'
+export const cancelDesc = 'Input canceled'
```
Comment (cjb) [inline shared/constants/provision.js]: confirm that the comment's still correct and the daemon changed their version?

Comment (chrisnojima) [reply shared/constants/provision.js]: yes, works on pass and paperkey

Comment (patrickxb) [reply shared/constants/provision.js]: we still have an error that has that string in it, but i'm not sure what `cancelDesc` is for.

Comment (patrickxb) [reply shared/constants/provision.js]: kex2/base.go:var ErrCanceled = errors.New("kex canceled by caller")

Comment (chrisnojima) [reply shared/constants/provision.js]: in my testing, we send this to cancel the login process when clicking back. it seems to function correctly on every step so i think we're good

**Pull request merged**